### PR TITLE
VDB-1368: add new geth patch as diff source

### PR DESCRIPTION
**Pull request by elizabethengelman**
companion PR in transformers repo to add an extractDiffs dockerfile: https://github.com/makerdao/vdb-mcd-transformers/pull/259

### CODE REVIEW

File: pkg/contract_watcher/types/method.go [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+	"fmt"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+// Method is our custom method struct
+type Method struct {
```
Comment (rmulhol) [inline pkg/contract_watcher/types/method.go]: Not sure I follow why this file is included in this PR?

File: libraries/shared/storage/fetcher/geth_rpc_storage_fetcher.go [hunk 2]
```diff
@@ -67,45 +64,31 @@ func (fetcher GethRpcStorageFetcher) FetchStorageDiffs(out chan<- types.RawDiff,
...
 
-func getAccountsFromDiff(stateDiff statediff.StateDiff) []statediff.AccountDiff {
-	accounts := append(stateDiff.CreatedAccounts, stateDiff.UpdatedAccounts...)
-	return append(accounts, stateDiff.DeletedAccounts...)
+	for _, account := range stateDiff.UpdatedAccounts {
+		logrus.Infof(processingDiffsLogString, len(account.Storage), common.Bytes2Hex(account.Key))
+		for _, accountStorage := range account.Storage {
+			rawDiff, formatErr := types.FromGethStateDiff(account, &stateDiff, accountStorage)
+			if formatErr != nil {
+				errs <- formatErr
```
Comment (rmulhol) [inline libraries/shared/storage/fetcher/geth_rpc_storage_fetcher.go:87]: do we potentially want to return after this? Not sure if sending an error will lead the caller to try and close the channel (leading to a panic on subsequent sends)

File: dockerfiles/extract_diffs/Dockerfile [hunk 3]
```diff
@@ -1,48 +0,0 @@
-FROM golang:alpine as builder
```
Comment (rmulhol) [inline dockerfiles/extract_diffs/Dockerfile:1]: Probably good to remove the [build/deploy](https://github.com/makerdao/vulcanizedb/blob/staging/.travis/deploy.sh#L32) here as well

File: go.mod [hunk 4]
```diff
@@ -35,6 +34,7 @@ require (
 	github.com/onsi/gomega v1.10.0
 	github.com/oschwald/maxminddb-golang v1.5.0 // indirect
 	github.com/pborman/uuid v1.2.0 // indirect
+	github.com/pkg/errors v0.8.1
```
Comment (rmulhol) [inline go.mod]: Is this required bc of Geth? For some reason I feel like at some point we accidentally required this at some point when we really wanted the core `errors` package

**Review by rmulhol — APPROVED** [review]
😤

File: pkg/contract_watcher/types/method.go (hunk 1 above)
Comment (elizabethengelman) [reply pkg/contract_watcher/types/method.go]: I think was probably added in due to a rebasing error. I'll remove it. ✂️

File: go.mod (hunk 4 above)
Comment (elizabethengelman) [reply go.mod]: 🤔  Not sure why that ended up in there, don't think it's necessary though.

File: dockerfiles/extract_diffs/Dockerfile (hunk 3 above)
Comment (elizabethengelman) [reply dockerfiles/extract_diffs/Dockerfile:1]: good catch!

File: libraries/shared/storage/fetcher/geth_rpc_storage_fetcher.go (hunk 2 above)
Comment (elizabethengelman) [reply libraries/shared/storage/fetcher/geth_rpc_storage_fetcher.go:87]: This piece is a little fuzzy for me - I think you're right, that if we were to send an error to the `errs` channel, then it's possible that the next accountStorage we're iterating through could try to send a diff (or an error) to a channel that's already been closed... I think. Also, it probably makes sense to stop processing diffs if the system is going to shut down anyway.

**Pull request merged**